editor/src/templates/image-dnd.spec.browser2.tsx

```diff
@@ -0,0 +1,403 @@
+      style={{
+        position: 'absolute',
+        width: 200,
+        height: 62,
+        top: 475,
+        left: 3,
+      }}
+      data-uid='imgimgimg'
+    />
+  </Storyboard>
+)
+`)
+  })
+
+  it('dragging from the "finder" works', async () => {
+    const newUID = 'imgimgimg'
+    FOR_TESTS_setNextGeneratedUid(newUID)
+
+    const editor = await renderTestEditorWithProjectContent(contents, 'await-first-dom-report')
+    const canvasControlsLayer = editor.renderedDOM.getByTestId(CanvasControlsContainerID)
+
+    const file = await makeImageFile(imgBase64, 'chucknorris.png')
+
+    const target = editor.renderedDOM.getByTestId('scene')
+    const targetBounds = target.getBoundingClientRect()
+
+    const endPoint = {
+      x: targetBounds.x + targetBounds.width / 2,
+      y: targetBounds.y + targetBounds.height / 2,
+    }
+
+    fireEvent(
+      canvasControlsLayer,
+      makeDragEvent('drag', canvasControlsLayer, { x: 5, y: 5 }, [file]),
+    )
+
+    fireEvent(canvasControlsLayer, makeDragEvent('drag', canvasControlsLayer, endPoint, [file]))
+
+    fireEvent(canvasControlsLayer, makeDragEvent('drop', canvasControlsLayer, endPoint, [file]))
+
+    await editor.getDispatchFollowUpActionsFinished()
+
+    await wait(1000) // read the image
+
+    expect(getPrintedUiJsCode(editor.getEditorState())).toEqual(`import * as React from 'react'
+import { Scene, Storyboard } from 'utopia-api'
+import { App } from '/src/app.js'
+import { View, Rectangle } from 'utopia-api'
+import { FlexRow } from 'utopia-api'
+
+export var storyboard = (
+  <Storyboard data-uid='0cd'>
+    <Scene
+      style={{
+        width: 700,
+        height: 759,
+        position: 'absolute',
+        left: 207,
+        top: 126,
+        paddingLeft: 91,
+      }}
+      data-testid='scene'
+      data-label='Playground'
+      data-uid='3fc'
+    />
+    <img
+      alt=''
+      src='./assets/chucknorris.png'
+      style={{
+        position: 'absolute',
+        left: 602.5,
+        top: 505.5,
+        width: 1,
+        height: 1,
+      }}
+      data-uid='imgimgimg'
+      data-aspect-ratio-locked
+    />
+  </Storyboard>
+)
+`)
+  })
+})
+
+// https://github.com/testing-library/react-testing-library/issues/339
+function makeDragEvent(
+  type: 'drag' | 'drop',
+  target: Element | Node,
+  clientCoords: { x: number; y: number },
+  fileList: Array<File>,
+): Event {
+  const opts = {
+    clientX: clientCoords.x,
+    clientY: clientCoords.y,
+    buttons: 1,
+    bubbles: true,
+    cancelable: true,
+  }
+  const fileDropEvent =
+    type === 'drop' ? createEvent.drop(target, opts) : createEvent.drag(target, opts)
+
+  Object.defineProperty(fileDropEvent, 'dataTransfer', {
+    value: {
+      getData: () => '',
+      items: fileList.map((f) => ({ kind: 'file', getAsFile: () => f })),
+      files: {
+        item: (itemIndex: number) => fileList[itemIndex],
+        length: fileList.length,
+      },
+    },
+  })
+
+  return fileDropEvent
+}
```

Comment:
This function living here is fine for now, but if we find ourselves needing to produce more drag and drop events in other tests we should definitely abstract part of it into `event-helpers.test-utils`

Reply:
moved it for good measure